Include/Python.h

```diff
@@ -34,6 +33,7 @@
 
 #include <assert.h>               // assert()
 #include <wchar.h>                // wchar_t
+#include <string.h>               // memcpy()
```

Comment:
It makes me sad to use memcpy() in the limited C API which should attempt to hide as many implementation details as possible. I don't get how the immortal object change interacts with the stable ABI. I would prefer to convert Py_INCREF/Py_DECREF to opaque function calls. It would avoid any C and C++ compatibility issue. At least, it would make the ABI a little bit more "forward compatible".

But that can be done later, since the priority is to unblock the next Python 3.12 beta release.

By the way, the nogil fork has also a completely different implementation which is also ABI incompatible unless these functions are implemented as opaque function calls.